src/Button.tsx

```diff
@@ -0,0 +1,192 @@
+
+  /**
+   * Which variant to display
+   *
+   * The options are as follows:
+   *
+   * - `normal`: A button with a border and a background
+   * - `simple`: No border or background (overrideable).
+   *     - This will require you adding an `:active` style in addition to `:hover`
+   */
+  variant?: "normal" | "simple";
+}
+
+/**
+ * Style system for Space Kit buttons
+ *
+ * This is intended to be used as an abstraction for your project's style guide.
+ * There are way too many configuration options to use this component directly.
+ *
+ * @see https://zpl.io/amdN6Pr
+ */
+export const Button: React.FC<Props> = ({
+  children,
+  disabled = false,
+  fab = false,
+  icon,
+  size = "default",
+  variant = "normal",
+  ...otherProps
+}) => {
+  /**
+   * Icon size in pixels
+   *
+   * This is stored so we can use the same value for `height` and `width`
+   */
+  const iconSize = size === "small" ? 12 : size === "large" ? 24 : 16;
+
+  const iconOnly = !children;
+
+  if (fab) {
+    if (!icon) {
+      throw new TypeError("FAB buttons are required to have an `icon`");
+    } else if (children) {
+      throw new TypeError("FAB buttons cannot have children, only an `icon`");
+    }
+  }
+
+  return (
+    <button
+      {...otherProps}
+      disabled={disabled}
+      css={[
+        {
+          "&[disabled]": {
+            backgroundColor: colors.silver.dark,
+            color: colors.grey.light,
+
+            // We need to also set the `:hover` on `:disabled` so it has a higher
+            // specificity than any `:hover` classes passed in. This also means
+            // that both of these need to be overriden if we want to use a custom
+            // disabled color.
+            ":hover": {
+              backgroundColor: colors.silver.dark,
+              color: colors.grey.light
+            }
+          }
+        },
+
+        variant === "simple" && { backgroundColor: "transparent" },
+        {
+          borderRadius: fab ? "100%" : 4,
+
+          borderWidth: 0,
+          ...(variant === "normal" && {
+            boxShadow:
+              "0 1px 4px 0 rgba(18, 21, 26, 0.08), inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)"
+          }),
+
+          minWidth: iconOnly
+            ? size === "small"
+              ? 28
+              : size === "large"
+              ? 42
+              : 36
+            : size === "small"
+            ? 76
+            : size === "large"
+            ? 112
+            : 100,
+
+          padding: size === "small" ? `5px 8px` : size === "large" ? 8 : 7,
+
+          ...(size === "small"
+            ? base.small
+            : size === "large"
+            ? base.large
+            : base.base),
+
+          fontWeight: 600,
+          // Disable the outline because we're setting a custom `:active` style
+          outline: 0,
+        },
+
+        // This must come after `:focus` or the `:focus` state will override `:active`
```

Comment:
We could write this like this to add more CSS specificity and avoid the condition mentioned in this comment:

```js
{
  '&:not([disabled])': {
    ':hover': ...
  }
}
```

Reply:
I mentioned this before; we this will make overrides with tailwind much more difficult because of that increased specificity.

Reply:
Oh yea, I forgot we had talked about that!